examples/evaluation_and_profiling/swe_bench/tests/test_iterative_predictor.py

```
@@ -0,0 +1,700 @@
+        await manager.cleanup()
+
+        assert len(manager.active_repos) == 0
+
+    @pytest.mark.asyncio
+    async def test_register_cleanup_error_handling(self):
+        """Test that register.py cleanup handles errors gracefully."""
+        from unittest.mock import AsyncMock
+        from nat_swe_bench.predictors.predict_iterative.tools.git_tool import RepoManager
+
+        # Create a mock that raises an exception
+        manager = RepoManager("/tmp/test")
```

Ruff (0.14.14)

Probable insecure usage of temporary file or directory: "/tmp/test"

(S108)